crates/swc_ecma_transforms_proposal/src/decorator_impl.rs

```diff
@@ -718,10 +717,10 @@ impl DecoratorPass {
             }
         }
         body.visit_mut_with(self);
-        c.visit_mut_with(self);
         c.ident = preserved_class_name.clone();
         replace_ident(&mut c.class, c.ident.to_id(), &preserved_class_name);
         c.class.body.extend(body);
+        c.visit_mut_with(self);
```

Comment:
This is wrong because of the time complexity. You should select the exact node before calling `visit_mut_with`

Reply:
sorry I don't get it, I don't see why moving this from line 721 to line 723 increases the time complexity

Reply:
Ah sorry, it's fine if it's about moving an existing line